onnxruntime/core/providers/cpu/tensor/upsample.cc

```diff
@@ -30,6 +35,46 @@ REGISTER_VERSIONED_TYPED_KERNEL(int32_t, 9, 9);
 REGISTER_VERSIONED_TYPED_KERNEL(int8_t, 9, 9);
 REGISTER_VERSIONED_TYPED_KERNEL(uint8_t, 9, 9);
 
+void UpsampleBase::AdjustOutputSizeAsPolicy(TensorShapeVector& output_dims, gsl::span<const int64_t> input_dims,
+                                            InlinedVector<float>& scales) const {
```

Comment:
May add an assert that keep_aspect_ratio_policy_ is one of STRETCH, NOT_LARGER or NOT_SMALLER. Otherwise, the result might not be expected.

Reply:
This seems to be done in `StringToKeepAspectRatioPolicy`